siliconcompiler/report/report.py

```diff
@@ -0,0 +1,402 @@
+
+    Args:
+        manifest(dictionary) : a dicitionary representing the manifest
+        key_search(string) : searches all keys for partial matches on this
+        string value_search(string) : searches all values for partial matches
+        on this string.
+
+    Example:
+        >>> search_manifest(jsonDict, key_search='input', value_search='v')
+        returns a filtered version of jsonDict where each path contains at
+        least one key that contains the substring input and has values that
+        contain v.
+    """
+    return_manifest = manifest
+    if key_search:
+        return_manifest = search_manifest_keys(return_manifest, key_search)
+    if value_search:
+        return_manifest = search_manifest_values(return_manifest, value_search)
+    return return_manifest
+
+
+def get_total_manifest_parameter_count(manifest):
+    """
+    Returns (int) the number of folders and files
+
+    Args:
+        manifest(dictionary) : a dicitionary representing the manifest
+        acc(int) : an accumulator of the current number of folders and files
+    """
+    acc = len(manifest)
+    for dictKeys in manifest:
+        if isinstance(manifest[dictKeys], dict):
+            acc += get_total_manifest_parameter_count(manifest[dictKeys])
+    return acc
+
+
+def get_metrics_source(chip, step, index):
+    """
+    returns a dictionary where the keys are files in the logs and reports for
+    a given step and index. The values are a list of the metrics that come from
+    that file. If a file is not in the dictionary, that implies that no metrics
+    come from it.
+
+    Args:
+        chip (Chip) : the chip object that contains the schema read from
+        step (string) : step of node
+        index (string) : index of node
+    """
+    file_to_metric = {}
+    tool, task = chip._get_tool_task(step, index)
+    metrics = chip.getkeys('tool', tool, 'task', task, 'report')
+    for metric in metrics:
+        sources = chip.get('tool', tool, 'task', task, 'report',
+                           metric, step=step, index=index)
+        for source in sources:
+            if source in file_to_metric:
+                file_to_metric[source].append(metric)
+            else:
+                file_to_metric[source] = [metric]
+    return file_to_metric
+
+
+def get_logs_and_reports(chip, step, index):
+    """
+    returns a list of 3-tuple that contain the path name of how to get to that
+    folder, the subfolders of that directory, and it's files. The list is
+    ordered by layer of directory.
+
+    Args:
+        chip (Chip) : the chip object that contains the schema read from
+        step (string) : step of node
+        index (string) : index of node
+    """
+    # could combine filters, but slighlty more efficient to seperate them
+    filters = {"inputs",
+               "reports",
+               "outputs",
+               f'{step}.log',
+               f'{step}.errors',
+               f'{step}.warnings'}
```

Comment:
if the filter is this broad, maybe we should just skip the filter and display all files?

Reply:
Sure